pkg/keyspace/tso_keyspace_group.go

```diff
@@ -552,6 +556,29 @@ func (m *GroupManager) SplitKeyspaceGroupByID(splitSourceID, splitTargetID uint3
 		if err = m.store.SaveKeyspaceGroup(txn, splitSourceKg); err != nil {
 			return err
 		}
+		// Iterate the keyspaces to update the config key.
+		keyspaceIDsToUnlock := make([]uint32, 0, len(keyspaces))
+		defer func() {
+			for _, id := range keyspaceIDsToUnlock {
+				m.km.metaLock.Unlock(id)
+			}
+		}()
+		for _, keyspace := range keyspaces {
```

Comment:
SaveKeyspaceMeta() needs to be called in batches too, otherwise the len(keyspaces) in a split needs to be <= maxTxnOps which isn't practical even in dev with tens thousands of keyspaces. Adding batching in another place increases complexity again.

Let's discuss about it today. @rleungx